internal/pkg/service/buffer/api/service/service.go

```diff
@@ -38,13 +38,14 @@ type service struct {
 }
 
 func New(d dependencies.ForServer) buffer.Service {
-	stats := statistics.New(d.Store(), d.Logger())
-	stats.Watch(d.Process().Ctx())
-
 	return &service{
 		deps:   d,
 		mapper: mapper.NewMapper(d.BufferAPIHost()),
-		stats:  stats,
+		stats: statistics.New(d.Process().Ctx(), func(ctx context.Context, stats []model.SliceStats) {
+			if err := d.Store().UpdateSliceStats(ctx, stats); err != nil {
+				d.Logger().Error("cannot update slice stats in etcd: %s", err.Error())
+			}
+		}),
 	}
```

Comment:
V teste sa neda inicilizovat `Manager` cez `new`, co moze priniest problem, ze to v teste inicializujes nejak inak ako v produkcii, napr. zabudnes nejaku hodnotu nastavit.

Je lepsie, ak je mozne aj v teste pouzit ten isty konstruktor.